src/tracing.rs

```diff
@@ -31,257 +39,117 @@ pub enum LoggerLevel {
-
-            // If it's not the first call to `install`, the
-            // `min_level` can be different. Let's update it.
-            tracing.min_level(min_level);
-
-            Ok(tracing)
-        }
-
-        /// Re-define the minimum logger level.
-        #[wasm_bindgen(setter, js_name = "minLevel")]
-        pub fn min_level(&self, min_level: LoggerLevel) {
-            let _ = self.handle.modify(|layer| layer.min_level = min_level.into());
-        }
-
-        /// Turn the logger on, i.e. it emits logs again if it was turned
-        /// off.
-        #[wasm_bindgen(js_name = "turnOn")]
-        pub fn turn_on(&self) {
-            let _ = self.handle.modify(|layer| layer.turn_on());
-        }
-
-        /// Turn the logger off, i.e. it no long emits logs.
-        #[wasm_bindgen(js_name = "turnOff")]
-        pub fn turn_off(&self) {
-            let _ = self.handle.modify(|layer| layer.turn_off());
-        }
-    }
-
-    impl From<LoggerLevel> for Level {
-        fn from(value: LoggerLevel) -> Self {
-            use LoggerLevel::*;
-
-            match value {
-                Trace => Self::TRACE,
-                Debug => Self::DEBUG,
-                Info => Self::INFO,
-                Warn => Self::WARN,
-                Error => Self::ERROR,
-            }
-        }
-    }
-
-    #[wasm_bindgen]
-    extern "C" {
-        #[wasm_bindgen(js_namespace = console, js_name = "debug")]
-        fn log_debug(message: String);
-
-        #[wasm_bindgen(js_namespace = console, js_name = "info")]
-        fn log_info(message: String);
+    level_filter_reload_handle: reload::Handle<LevelFilter, JsLoggingSubscriber>,
+}
 
-        #[wasm_bindgen(js_namespace = console, js_name = "warn")]
-        fn log_warn(message: String);
+/// Type to install and to manipulate the tracing layer.
+#[wasm_bindgen]
+pub struct Tracing {
+    inner: Arc<Mutex<TracingInner>>,
+}
 
-        #[wasm_bindgen(js_namespace = console, js_name = "error")]
-        fn log_error(message: String);
+impl fmt::Debug for Tracing {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Tracing").finish_non_exhaustive()
     }
+}
 
-    struct Layer {
-        min_level: Level,
-        enabled: bool,
+#[wasm_bindgen]
+impl Tracing {
+    /// Check whether the `tracing` feature has been enabled.
+    #[wasm_bindgen(js_name = "isAvailable")]
+    pub fn is_available() -> bool {
+        true
     }
 
-    impl Layer {
-        fn new<L>(min_level: L) -> Self
-        where
-            L: Into<Level>,
-        {
-            Self { min_level: min_level.into(), enabled: true }
-        }
-
-        fn turn_on(&mut self) {
-            self.enabled = true;
-        }
-
-        fn turn_off(&mut self) {
-            self.enabled = false;
-        }
+    fn install_or_get_inner() -> Arc<Mutex<TracingInner>> {
+        static mut INSTALL: Option<Arc<Mutex<TracingInner>>> = None;
+        static INSTALLED: Once = Once::new();
```

Comment:
`OnceLock` is stable, can you not use that instead of a separate `Once` and `static mut`?

Reply:
probably, this bit was cargo-culted from the previous impl! Will take a look.